examples/with-apollo-and-redux/lib/initApollo.js

```diff
@@ -0,0 +1,36 @@
+import { ApolloClient, createNetworkInterface } from 'react-apollo'
+import fetch from 'isomorphic-fetch'
+
+let apolloClient = null
+
+// Polyfill fetch() on the server (used by apollo-client)
+if (!process.browser) {
+  global.fetch = fetch
+}
+
+function create () {
+  return new ApolloClient({
+    ssrMode: !process.browser, // Disables forceFetch on the server (so queries are only run once)
+    networkInterface: createNetworkInterface({
+      uri: 'https://api.graph.cool/simple/v1/cixmkt2ul01q00122mksg82pn', // Server URL (must be absolute)
+      opts: { // Additional fetch() options like `credentials` or `headers`
+        credentials: 'same-origin'
+      }
+    })
+  })
+}
+
+export default function initApollo () {
+  // Make sure to create a new client for every server-side request so that data
+  // isn't shared between connections (which would be bad)
```

Comment:
In theory the constructed `apollo` and `redux` constants created server-side in withData's getInitialProps could be returned in the resulting JSON, but need to make sure that it's only passed from getInitialProps to the render when server-side, and also need to check how it would serialize (although such a serialization isn't necessary server-side as everything is in memory).

Also see https://github.com/zeit/next.js/issues/978

Anyway, there might be something to improve here, but don't know if worth the effort